app/models/user.rb

```diff
@@ -1,17 +1,23 @@
 class User < ActiveRecord::Base
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable and :omniauthable
+  devise :database_authenticatable, # :registerable,
+         :recoverable, :rememberable, :trackable, :validatable
 
   VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-]+(\.[a-z]+)*\.[a-z]+\z/i
```

Comment:
The quality of mail validation is debatable.  I opted to go with most of Devise's validations, except for length (Devise allows emails longer than RFC 3696, errata ID 1690 allows.  I left a couple of tests in the suite that test devise validations for sanity sake.